src/meta/src/controller/catalog/alter_op.rs

```diff
@@ -782,4 +784,63 @@ impl CatalogController {
 
         Ok(version)
     }
+
+    // drop table associated source is a special case of drop relation, which just remove the source object and associated state table, keeping the streaming job and fragments.
+    pub async fn drop_table_associated_source(
+        txn: &DatabaseTransaction,
+        drop_table_connector_ctx: &DropTableConnectorContext,
+    ) -> MetaResult<(Vec<PbUserInfo>, Vec<PartialObject>)> {
+        let to_drop_source_objects: Vec<PartialObject> = Object::find()
+            .filter(object::Column::Oid.is_in(vec![drop_table_connector_ctx.to_remove_source_id]))
+            .into_partial_model()
+            .all(txn)
+            .await?;
+        let to_drop_internal_table_objs: Vec<PartialObject> = Object::find()
+            .select_only()
+            .columns([
+                object::Column::Oid,
+                object::Column::ObjType,
+                object::Column::SchemaId,
+                object::Column::DatabaseId,
+            ])
```

Comment:
Is it the same as above?
```suggestion
            .into_partial_model()
```

Reply:
Not sure about it, cc @yezizp2012

Reply:
seems it is equivalent.